### DISCUSSION TITLE
Question about the implementation of GAT

### DISCUSSION THREAD

**Discussion by ZikangZhou** [pyg-team/pytorch_geometric#2187] [Q&A]
I'm confused about the x_l and x_r in GAT. When calling propagate(), it passes an argument x=(x_l, x_r), which is a tuple. I guess one of the elements in this tuple is for the x_i in message() later, and the other one is for the x_j.  But which one corresponds to the source nodes and which one corresponds to the target nodes?

**Comment by rusty1s** [comment]
The left one corresponds to source nodes and the right one corresponds to target nodes. You can think of it as passing `x_l` to `x_r`.

**Reply by ZikangZhou** [reply]
Thank you, got it.
